atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderRole.java

```diff
@@ -710,9 +710,12 @@ private boolean isEntryConsistent(final long newEntryPosition) {
       } while (index > 0 && lastEntry != null && lastEntry.type() != ZeebeEntry.class);
     }
 
-    return lastEntry == null
-        || lastEntry.type() != ZeebeEntry.class
-        || newEntryPosition > ((ZeebeEntry) lastEntry.entry()).highestPosition();
+    long lastPosition = -1;
```

Comment:
We can pass the last entry to the validator instead of the last position but I we just use what's in memory, the validation won't work after a leader election, right? I agree that the Raft impl knows too much about zeebe but I think that's a consequence of mixing both as this issue does explicitly.

Reply:
The followers are also appending; furthermore on opening/restart we going through the entire log to build the index

Reply:
The follower role just appends entries without knowing the type of entries it's appending, right? We still have to check if the entry is a zeebe entry before setting it so that coupling is still there. But I'll make that change

Reply:
We could do a pair programming session today to finish this topic